code/gui/actuatorsetup.cpp

```diff
@@ -0,0 +1,103 @@
+#include "actuatorsetup.h"
+#include "ui_actuatorsetup.h"
+#include "../controller/actuator.h"
+#include <QtCore>
+
+ActuatorSetup::ActuatorSetup(std::shared_ptr<Actuator>& controller, QWidget *parent) :

```

Comment:
Going to skip most of the GUI stuff because it's hard to tell whether it's right or not without working directly with it, and also it's very obvious if there's an error when running the code.